src/libraries/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -93,6 +98,9 @@ public HashSet(IEqualityComparer<T>? comparer)
             }
 
             _comparer = comparer;
+#if BIT64
+            _fastModMultiplier = HashHelpers.GetFastModMultiplier(0);
```

Comment:
Is the JIT able to transform this into `ulong.MaxValue` directly?

Reply:
Thanks for the review!  I'm not sure how I could go about verifying this or forcing that behavior.  Any suggestions?

Reply:
It would be a division by zero so it shouldn't. It should not be initialized here. Just like Dictionary, HashSet allocates its buckets lazily, and until that is done, the multiplier is not needed. The multiplier should be set/updated every place that a new `_buckets` array is allocated -- looks like there are 4 places, if you search for `new int[`

Reply:
It's not a div by zero (it does `ulong.MaxValue / (divisor + 1)`).  I think that it should considering `GetFastModMultiplier()` is defined with `=>` and thus a good target for aggressive inlining?

Reply:
(To answer myself: yes, [it does inline to the point it becomes just `ulong.MaxValue`](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHuGBXAGwgA7AOYMAZhAgBuLj3qsAFAEoOcnj0kQGAXlbZ+YAdgwwAyhl7jxAdQCWGABYWr4xQAZls6hoC+6jQYAhWJyJD5BUX1DY1MXa3sneLcBYTFcS2tVHQA+CLSAOgBZbAQANQNeGAYAegZFDNcGAGoGci8aXyA==), but @danmosemsft's suggestion seems better.)

Reply:
It's `ulong.MaxValue / divisor + 1` not `ulong.MaxValue / (divisor + 1)`
https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Collections/HashHelpers.cs#L95